lib/kernel.ts

```diff
@@ -155,64 +185,44 @@ const getSelected = (schema: JsonSchema): { links: any; properties: any } => {
 	};
 };
 
-const rectifySelected = (
-	selected: any,
-	filteredSelected: { [x: string]: any },
-) => {
-	for (const [key, value] of Object.entries(selected)) {
-		if (key in filteredSelected) {
-			rectifySelected(value, filteredSelected[key]);
+// TODO: rename with something more descriptive.
+const rectifySelectObject = (
+	selectObject: SelectObject,
+	authorizedSelectObject: SelectObject,
+): SelectObject => {
+	const result = { ...selectObject };
+
+	for (const [key, value] of Object.entries(result)) {
+		if (key in authorizedSelectObject) {
+			rectifySelectObject(value, (authorizedSelectObject as any)[key]);
 		} else {
-			Reflect.deleteProperty(selected, key);
+			Reflect.deleteProperty(result, key);
 		}
 	}
-};
-
-const getQueryFromSchema = async (
-	context: Context,
-	backend: DatabaseBackend,
-	session: string,
-	schema: JsonSchema | ViewContract,
-	mask?: JsonSchema,
-) => {
-	// TS-TODO: Refactor this to avoid type coercion
-	let finalSchema: JsonSchema = (
```

Comment:
Looks like this is missing now

Reply:
It was moved to `utils.preprocessQuerySchema()`.

Reply:
Suggestions for the name of the function are welcome :)

Reply:
The view lookup?

Reply:
Yes. I wouldn't mind doing this explicitly in `kernel.ts`, but it's being used three or four times, and that would result in some code duplication.

Reply:
Ah it's fine the way you did it.